### ISSUE TITLE
It is OK to train RUBER and the model to be evaluated with the same dataset?

### ISSUE THREAD

**Issue by cgsdfc** [liming-vie/RUBER#8]
The setting is like this:
1. I train RUBER so that it can evaluate my model, say M.
2. I train M so that it can be evaluated by RUBER.
3. I have a dataset spilt in train, test and valid subsets.

My question is: should the training sets of RUBER and M be different?
Should M be trained on the train set while RUBER be trained on test set?
Thanks!

**Comment by liming-vie** [comment]
For this question I'm not sure...
In my experiment, I use the same setting as you did.
